### PR TITLE
Document OCP on OSP with OVNKubernetes networkType

### PR DESCRIPTION
**Pull request by EmilienM**
OVNKubernetes is a supported backend that enables more use-cases for our
community deploying OCP on OSP.

This patch creates a new page that will explain step by step how to
deploy OCP on OSP with the OVNKubernetes networkType.

Note: the page was inspired from Kuryr, as it's also documented another
networkType.

Jira: https://issues.redhat.com/browse/OSASINFRA-2234

### CODE REVIEW

**Comment by EmilienM** [comment]
@maxwelldb
ping, my first PR here, feel free to take over. I'm on parental leave until February 1st.

Thanks!

File: installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
+[id="installing-openstack-installer-kuryr"]
+= Installing a cluster on OpenStack with OVNKubernetes
```
Comment (maxwelldb) [inline installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc:2]:
```suggestion
= Installing a cluster on OpenStack with OVN-Kubernetes
```

**Comment by maxwelldb** [comment]
@EmilienM Which OCP versions should this apply to, and is there a QE contact the Jira ticket could be assigned to?

**Comment by EmilienM** [comment]
> @EmilienM Which OCP versions should this apply to, and is there a QE contact the Jira ticket could be assigned to?

4.6 and beyond / not AFIK
ping @pierreprinetti : who from QE can help here?

File: installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc (hunk 1 above)
Comment (EmilienM) [reply installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc:2]: OVNKubernetes is the networkType, do you still want to rename it for the page title?

Comment (maxwelldb) [reply installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc:2]: That's my thought. "OVN-Kubernetes" seems preferred in the titles of other Red Hat docs.

Comment (EmilienM) [reply installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc]: ack, done.

**Comment by maxwelldb** [comment]
@EmilienM No rush, as I'll probably end up publishing this after 4.7 GAs, but the reason that Kuryr necessitates separate assemblies is that using it changes a lot about the 'default,' non-Kuryr flows. 

Is it likely that OVN-K8s would do the same here, eventually? 

I'm trying to figure out how this fits into the docs (does it definitely need to be its own assembly? probably :smile:) and what the maintenance cost will be.

**Comment by EmilienM** [comment]
> @EmilienM No rush, as I'll probably end up publishing this after 4.7 GAs, but the reason that Kuryr necessitates separate assemblies is that using it changes a lot about the 'default,' non-Kuryr flows.

ack

> Is it likely that OVN-K8s would do the same here, eventually?

yes, I guess.

> I'm trying to figure out how this fits into the docs (does it definitely need to be its own assembly? probably smile) and what the maintenance cost will be.

The patch was pushed in the shiftstack fork, feel free to take over it or ask anyone from our team to push -f. I'm away for the next weeks.

**Comment by maxwelldb** [comment]
For me, later: https://github.com/shiftstack/openshift-docs/tree/osp/ovn

**Comment by EmilienM** [comment]
@maxwelldb it was already deferred in 4.7 and now we're deferring again in 4.8?

**Comment by maxwelldb** [comment]
@EmilienM OSASINFRA-2234 was not an acked part of 4.7, nor is it in 4.8 at the moment. The issue currently doesn't have a fix version or parent epic.

The issue here is is less the challenge that it raises for CCS and more the complications that it can cause for QE and Support. Any docs that we add have to be QEed and backed by Support, and those things can have legal ramifications. 

So, to proceed, I'm trying to figure out a few things:

A. Does this change describe a feature or process that was added in <=4.7? If so, is there a px-, qe-, and docs-acked issue for that process or feature? 
B. Is this doc for a _new_ feature or process that must be backported from a future release? 
C. Is this doc for a new feature or process that we can publish just in a future release? 

If A, this change is really a bug fix. We can open a BZ and process it that way. 

If B, this is a major change that needs to go through our change management process--a bunch of acks, announcements, etc. QE, PX, etc. are free to reject the change.

If C, we can put it on the docket for a future release like any other work. 

Does that make sense?

**Comment by EmilienM** [comment]
ack, I think we're in the C case now. Thanks

**Comment by maxwelldb** [comment]
@EmilienM Oh, yikes, forgot about this one. Is this something you'd like to look at including for 4.10?

File: installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+= Installing a cluster on OpenStack with OVN-Kubernetes
+include::modules/common-attributes.adoc[]
+:context: installing-openstack-installer-ovnk8s
+
+toc::[]
+
+In {product-title} version {product-version}, you can install a customized cluster on
+{rh-openstack-first} that uses Open Virtual Networking with Kubernetes. To customize the installation, modify parameters in the `install-config.yaml` before you install the cluster.
+
+== Prerequisites
```
Comment (maxwelldb) [inline installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc:11]:
```suggestion
[id="prerequisites_installing-openstack-installer-ovnk8s"]
== Prerequisites
```

File: installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+include::modules/ssh-agent-using.adoc[leveloffset=+1]
+include::modules/installation-osp-accessing-api.adoc[leveloffset=+1]
+include::modules/installation-osp-accessing-api-floating.adoc[leveloffset=+2]
+include::modules/installation-osp-accessing-api-no-floating.adoc[leveloffset=+2]
+include::modules/configuring-hybrid-ovnkubernetes.adoc[leveloffset=+1]
+include::modules/installation-launching-installer.adoc[leveloffset=+1]
+include::modules/installation-osp-verifying-cluster-status.adoc[leveloffset=+1]
+include::modules/cli-logging-in-kubeadmin.adoc[leveloffset=+1]
+
+== Next steps
```
Comment (maxwelldb) [inline installing/installing_openstack/installing-openstack-installer-ovnk8s.adoc:40]:
```suggestion
[id="next-steps_openstack-installer-ovnk8s"]
== Next steps
```

**Comment by EmilienM** [comment]
@maxwelldb changes were applied, thanks for the review!

**Comment by maxwelldb** [comment]
If this does need inclusion (see discussion in Jira), I'd say that I can't contain it for 4.10 any longer. 

If the ShiftStack team wants to include it as a "new" docs-only feature for 4.11, though, that would be fine by me. The docs are close to done.

**Comment by bobfuru** [comment]
In preparation for OCP 4.10 GA, I'm moving the Milestone to "Next Release" as this PR did not make the merge by the 4.10 GA date. Post-GA, any open 4.10 PRs fall under the "Next Release" bucket in the same way that 4.6-4.9 PRs are already there. This change does not have any impact on the work in this PR; it's a housekeeping task to keep account of all PRs that had already merged by the 4.10 GA date. 😁

**Comment by maxwelldb** [comment]
Closing per Jira. Re-open if needed.

**Pull request closed without merging**